rotkehlchen/api/rest.py

```diff
@@ -3358,6 +3363,17 @@ def decode_pending_ethereum_transactions(
             async_query: bool,
             evm_addresses: Optional[list[ChecksumEvmAddress]],
     ) -> Response:
+        """
+        This method should be called after querying ethereum transactions and does the following:
+        - Query missing receipts
+        - Decode ethereum transactions
+
+        It can be a slow process and this is why it is important to set the list of addresses
+        queried per module that need to be decoded.
+
+        This logic is executed by the frontend in pages where the set of transactions needs to be
```

Comment:
Question: if this method is to be only queried by the front-end, does it not also query the ethereum transactions? Or does the fronted have to make sure transactions are queried? Why not keep it simple and do it all in one place? Or perhaps I am misunderstanding

Reply:
We have this as two queries since the fronted reuses the same logic they have for the ethereum transactions view. So the ethereum transaction query logic is not only used here. What is only used in this page is the `decode all pending transactions`